crates/router/src/core/payments/helpers.rs

```diff
@@ -6102,3 +6102,36 @@ pub fn validate_platform_fees_for_marketplace(
         }
     }
 }
+
+pub fn get_optional_card_type_from_additional_payment_method_data(
```

Comment:
instead of a helper function, write a impl based function on AdditionalPaymentData.